### ISSUE TITLE
empty list should sort to another empty list, not (undef)

### ISSUE THREAD

**Issue by garu** [CaballerosTeam/Sort-Naturally-XS#3]
Hi! First of all, let me thank you again for writing and maintaining Sort::Naturally::XS. I've been making some tests with it and it looks like it really speed things up!

The issue here is with the return value of nsort. Whenever it gets an empty list, instead of returning another empty list (e.g. `()`) it is returning `undef`, which behaves as a single element when in list context. This is a very common issue with list/scalar context!

> `perl -MSort::Naturally::XS -E 'my @l; say scalar @l; my @o = nsort(@l); say scalar @o'`
0
1

So if I do (a common case, I suspect):

```perl
   @list = nsort( @list );
```

and `@list` is empty (e.g `()`), then after the call to `nsort` it will not be empty anymore, and instead contain 1 element: `(undef)`. In fact:

```
> perl -Ilib  -MSort::Naturally::XS -E 'my @l; @l = nsort(@l); use DDP; p @l'
[
  [0]  undef
]
```

The solution is returning the empty list `()`, which in scalar context is `undef` and in list context is, well, an actual empty list :)

Hope this helps! Thanks again for writing such an interesting module!

**Comment by CaballerosTeam** [comment]
Fixed in 127df11e8beae59c345b8d2ec2d25c736d467b24

**Comment by CaballerosTeam** [comment]
https://github.com/CaballerosTeam/Sort-Naturally-XS/releases/tag/v0.7.9
